android/src/main/java/com/rt2zz/reactnativecontacts/ContactsManager.java

```diff
@@ -130,9 +131,140 @@ public void run() {
+        name.put(ContactsContract.CommonDataKinds.StructuredName.FAMILY_NAME, familyName);
+        name.put(ContactsContract.CommonDataKinds.StructuredName.MIDDLE_NAME, middleName);
+        name.put(ContactsContract.CommonDataKinds.StructuredName.PREFIX, prefix);
+        name.put(ContactsContract.CommonDataKinds.StructuredName.SUFFIX, suffix);
+        contactData.add(name);
+
+        ContentValues organization = new ContentValues();
+        organization.put(ContactsContract.Data.MIMETYPE, Organization.CONTENT_ITEM_TYPE);
+        organization.put(Organization.COMPANY, company);
+        organization.put(Organization.TITLE, jobTitle);
+        organization.put(Organization.DEPARTMENT, department);
+        contactData.add(organization);
+
+        for (int i = 0; i < numOfEmails; i++) {
+            ContentValues email = new ContentValues();
+            email.put(ContactsContract.Data.MIMETYPE, CommonDataKinds.Email.CONTENT_ITEM_TYPE);
+            email.put(CommonDataKinds.Email.TYPE, emailsLabels[i]);
+            email.put(CommonDataKinds.Email.ADDRESS, emails[i]);
+            contactData.add(email);
+        }
+
+        for (int i = 0; i < numOfPhones; i++) {
+            ContentValues phone = new ContentValues();
+            phone.put(ContactsContract.Data.MIMETYPE, CommonDataKinds.Phone.CONTENT_ITEM_TYPE);
+            phone.put(CommonDataKinds.Phone.TYPE, phonesLabels[i]);
+            phone.put(CommonDataKinds.Phone.NUMBER, phones[i]);
+            contactData.add(phone);
+        }
+
+        for (int i = 0; i < numOfPostalAddresses; i++) {
+            ContentValues structuredPostal = new ContentValues();
+            structuredPostal.put(ContactsContract.Data.MIMETYPE, CommonDataKinds.StructuredPostal.CONTENT_ITEM_TYPE);
+            structuredPostal.put(CommonDataKinds.StructuredPostal.STREET, postalAddressesStreet[i]);
+            structuredPostal.put(CommonDataKinds.StructuredPostal.CITY, postalAddressesCity[i]);
+            structuredPostal.put(CommonDataKinds.StructuredPostal.REGION, postalAddressesRegion[i]);
+            structuredPostal.put(CommonDataKinds.StructuredPostal.COUNTRY, postalAddressesCountry[i]);
+            structuredPostal.put(CommonDataKinds.StructuredPostal.POSTCODE, postalAddressesPostCode[i]);
+            //No state column in StructuredPostal
+            //structuredPostal.put(CommonDataKinds.StructuredPostal.???, postalAddressesState[i]);
+            contactData.add(structuredPostal);
+        }
+
+        Intent intent = new Intent(Intent.ACTION_INSERT, ContactsContract.Contacts.CONTENT_URI);
+        intent.putExtra(ContactsContract.Intents.Insert.NAME, displayName);
+        intent.putParcelableArrayListExtra(ContactsContract.Intents.Insert.DATA, contactData);
+        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+
+        Context context = getReactApplicationContext();
+        context.startActivity(intent);
+
+    }
+    /*
+         * Adds contact to phone's addressbook
```

Comment:
If you place this on the same indentation as it was the diff will not pick it up. Please undo the additional indentation on the comment.